### DISCUSSION TITLE
Missing resources in product build

### DISCUSSION THREAD

**Discussion by rheydenr** [eclipse-tycho/tycho#3773] [Q&A]
Hi all, I'm using Tycho for a long time. Recently, I've updated to version 4.0.4 and stated, that no resource fragments are bundled into final product. They are built correctly in their target directory, but none of them appears in the product. Are there any changes (I'm coming from version 2) to Tycho which prevents packaging of plugins which doesn't contain Java classes? The build is running successful without any errors. Thanks in advance for your answers, Ralf.
PS: I didn't find any mailing list or forum to post my question, therefore I used this channel :-) )

**Comment by laeubi** [comment]
Can you provide an [integration-test](https://github.com/eclipse/tycho/wiki#providing-an-integration-test) to demonstrate the issue? Are you using a feature based product or a plugin based product?

**Reply by rheydenr** [reply]
I'm using plugin based product. I'll try to create a test case. Thanks for your fast reply.

**Reply by rheydenr** [reply]
So I've created a simple test project under https://github.com/rheydenr/rcptestprojekt
The problem is that I'm currently unable to create even a simple RCP project which creates a stand-alone application. Nevertheless, the test project contains all necessary stuff to demonstrate my issue. The "icons" project is built but not added resulting archive. Any hints are welcome.

**Reply by laeubi** [reply]
@rheydenr I can't see that the project includes the icon project here

https://github.com/rheydenr/rcptestprojekt/blob/5be0cf6a448d2a96f158085822f408fe6117e2db/de.rhe.rcptestprojekt.updatesite/de.rhe.rcptestprojekt.product#L27-L30

Also the dependency does not require the icon project as far as I can see

https://github.com/rheydenr/rcptestprojekt/blob/5be0cf6a448d2a96f158085822f408fe6117e2db/de.rhe.rcptestprojekt/META-INF/MANIFEST.MF#L7-L16

therefore it is not included in the final product.

**Reply by rheydenr** [reply]
Ok, I thought the dependency is included because it's a fragment bundle. So I cannot add this fragment as a dependency (it's not in the list of available plugins). 

So, I've updated the product file and added the icons plugin:
`   <plugins>
      <plugin id="de.rhe.rcptestprojekt"/>
      <plugin id="de.rhe.rcptestprojekt.icons"/>
      <plugin id="org.eclipse.jface"/>
   </plugins>
`
Now, the build is failing, yet:
`
Missing requirement: de.rhe.rcptestprojekt.product 1.0.0.qualifier requires 'org.eclipse.equinox.p2.iu; org.eclipse.equinox.executable.feature.group 0.0.0' but it could not be found
`
I've really don't know where to start to fix this issue. Any hints are welcome.

**Reply by laeubi** [reply]
> Missing requirement: de.rhe.rcptestprojekt.product 1.0.0.qualifier requires 'org.eclipse.equinox.p2.iu; org.eclipse.equinox.executable.feature.group 0.0.0' but it could not be found

You either need to include the equinox launcher in your setup or disable the "this product includes native lauchers" option.

**Reply by rheydenr** [reply]
Ok, I've checked this. Now, the resulting product contains all jars that I've expected. Unfortunately, the generated `config.ini` doesn't contain the plugin. So, I've checked this in the Tycho demo project and there's the same error. I've built the `tycho/demo/itp04-rcp` project (in branch tycho-4.0.x). But this application cannot be started (neither in the IDE nor from the resulting product file). The error log in the workspace contains a huge count of several exceptions.

Now, my question is: How can I achieve that the application plugin is written into the config.ini?

**Reply by laeubi** [reply]
Does it need to be in config.ini at all? Have you checked with your actual prduct? The config ini itself is not written by Tycho but by P2.

**Reply by rheydenr** [reply]
I think it's necessary to start the main plugins. Currently, I've run into failure because of missing / not active plugins. But for the moment, I think the origin question is answered. Follow-up questions are then in a new thread :-) Thanks for your fast help.

**Reply by laeubi** [reply]
Fragments can not be started as they are only attached to their host.
